hicommonwealth/commonwealth
# Allow user to buy stake without requiring community membership

## Link to Issue
Closes: 
- https://github.com/hicommonwealth/commonwealth/issues/6587
- https://github.com/hicommonwealth/commonwealth/issues/6594

## Description of Changes
- Allow user to buy stake in a community they are not a member of.
- Joins user address used to buy stake, to that community after they buy stake.

## "How We Fixed It"
N/A

## Test Plan
For local testnet
1. Add Sepolia chain node to db
2. Have 2 wallet addresses, each with some faucet Sepolia 
3. Switch your eth wallet network to `Sepolia` and login (address 1)
4. Create a new community with stake enabled. **Important**: in network selection, select 'Sepolia' as your community network. Once the community is created, you will already be a member of that community. 
5. Log out, go to any non-community page (ex: `/dashboard`), and use a different eth address (address 2) to log in (make sure the network is `Sepolia`). 
6. Now go to the community you created with address 1, verify you are not a member of the community (membership can be removed from the profile page)
7. From the vote weight module in the community sidebar, click on the 'Buy stake' button
8. From the stake modal, buy some stakes and verify the whole process works smoothly without errors, verify that the new vote/stake weight is reflected in the community sidebar, and after the page reload
9. From the vote weight module in the community sidebar, click on the 'Sell stake' button
8. From the stake modal, sell some stakes and verify the whole process works smoothly without errors, verify that the new vote/stake weight is reflected in the community sidebar, and after the page reload
9. After buying a stake from any page (global explore, active community, community directory), verify that you become a member of that community with the address you used to buy the stake. (this will only happen if you are not a member of that community)
10. Log out, go to any non-community page, and log in with non-evm wallet. Go to any evm based community with stake enabled, verify that you see the 'buy stake' button disabled with the tooltip `Connect an ethereum based wallet or join this community to buy stake`. This should happen on all pages where you can buy stake from i.e global explore, community directory, active community.


## Deployment Plan
N/A

## Other Considerations
N/A
